com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -74,23 +75,24 @@ public Task<ShaderPreprocessor> create(IResource input) throws IOException, Comp
             .addInput(input);
 
         // Parse source for includes and add the include nodes as inputs/dependancies to the shader
-        String source     = new String(input.getContent(), StandardCharsets.UTF_8);
-        String projectDir = this.project.getRootDirectory();
+        String source = new String(input.getContent(), StandardCharsets.UTF_8);
         ShaderPreprocessor shaderPreprocessor = new ShaderPreprocessor(this.project, input.getPath(), source);
         String[] includes = shaderPreprocessor.getIncludes();
 
         for (String path : includes) {
             taskBuilder.addInput(this.project.getResource(path));
         }
 
+        String platformString = this.project.getPlatformStrings()[0];
+
         // Include the spir-v flag into the cache key so we can invalidate the output results accordingly
-        String spirvCacheKey = "output_spirv=" + getOutputSpirvFlag();
+        String shaderCacheKey = String.format("output_spirv=%s;platform_key=%s", getOutputSpirvFlag(), platformString);
+
         taskBuilder.addOutput(input.changeExt(params.outExt()));
         taskBuilder.setData(shaderPreprocessor);
-        taskBuilder.addExtraCacheKey(spirvCacheKey);
+        taskBuilder.addExtraCacheKey(shaderCacheKey);
```

Comment:
This doesn't work as well as I thought it would 😢 I'll leave it here for now anyway

Reply:
maybe it's better to add comment into code then (just to know it doesn't work or work in unexpected way) to avoid misunderstanding when you think that some functionality is ON when it isn't